### PR TITLE
[TA] Added offset and length support in CatorizedEntities and LinkedEntitiesMatch

### PR DESCRIPTION
**Pull request by mssfang**
- Added "Offset" and "length" support in CatorizedEntities and LinkedEntitiesMatch

- Additional, refactor other endpoints tests exception sentiment analysis. It will be handle in  #13855

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/LinkedEntityMatch.java [hunk 1]
```diff
@@ -51,4 +79,22 @@ public double getConfidenceScore() {
     public String getText() {
         return this.text;
     }
+
+    /**
+     * Get the offset of entity text.
```
Comment (iscai-msft) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/LinkedEntityMatch.java:84]: @maririos made this comment on my PR too, but it would be clearer if this was "Get the offset of the linked entity match text"

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientTestBase.java [hunk 2]
```diff
@@ -484,6 +521,8 @@ static void validateCategorizedEntity(
         CategorizedEntity expectedCategorizedEntity, CategorizedEntity actualCategorizedEntity) {
         assertEquals(expectedCategorizedEntity.getSubcategory(), actualCategorizedEntity.getSubcategory());
         assertEquals(expectedCategorizedEntity.getText(), actualCategorizedEntity.getText());
+        assertEquals(expectedCategorizedEntity.getOffset(), actualCategorizedEntity.getOffset());
```
Comment (iscai-msft) [inline sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientTestBase.java:524]: next time can you make these other test changes in another PR, it's a little hard to see what actual tests were added for offset and length

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/CategorizedEntity.java [hunk 3]
```diff
@@ -43,6 +53,28 @@ public CategorizedEntity(String text, EntityCategory category, String subcategor
...
+     * Creates a {@link CategorizedEntity} model that describes entity.
+     *
+     * @param text The entity text as appears in the request.
+     * @param category The entity category, such as Person/Location/Org/SSN etc.
+     * @param subcategory The entity subcategory, such as Age/Year/TimeRange etc.
+     * @param offset The start position for the entity text.
+     * @param length The length for the entity text.
+     * @param confidenceScore A confidence score between 0 and 1 of the extracted entity.
+     */
+    public CategorizedEntity(String text, EntityCategory category, String subcategory, int offset, int length,
```
Comment (maririos) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/CategorizedEntity.java:70]: should we keep the `offset` and `length` after `confidenceScore` so that someone that depended on previous constructor would use this by just appending properties?

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/LinkedEntityMatch.java [hunk 4]
```diff
@@ -51,4 +79,22 @@ public double getConfidenceScore() {
...
+     *
+     * @return The offset of entity text.
+     */
+    public int getOffset() {
+        return offset;
+    }
+
+    /**
+     * Get the length of entity text.
```
Comment (maririos) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/LinkedEntityMatch.java:93]: linked entity text...

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java [hunk 5]
```diff
@@ -144,22 +144,24 @@
...
      */
     static DetectLanguageResultCollection getExpectedBatchDetectedLanguages() {
-        DetectedLanguage detectedLanguage1 = new DetectedLanguage("English", "en", 0.0, null);
-        DetectedLanguage detectedLanguage2 = new DetectedLanguage("Spanish", "es", 0.0, null);
-        DetectedLanguage detectedLanguage3 = new DetectedLanguage("(Unknown)", "(Unknown)", 0.0, null);
-
-        TextDocumentStatistics textDocumentStatistics1 = new TextDocumentStatistics(26, 1);
-        TextDocumentStatistics textDocumentStatistics2 = new TextDocumentStatistics(40, 1);
-        TextDocumentStatistics textDocumentStatistics3 = new TextDocumentStatistics(6, 1);
+        final TextDocumentBatchStatistics textDocumentBatchStatistics = new TextDocumentBatchStatistics(3, 3, 0, 3);
```
Comment (maririos) [inline sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java:147]: is this just cleanup? or what is the motivation for these changes

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java [hunk 6]
```diff
@@ -217,32 +218,40 @@ static RecognizeEntitiesResult getExpectedBatchCategorizedEntities2() {
...
+                    new LinkedEntityCollection(new IterableStream<>(getLinkedEntitiesList2()), null)));
+        return new RecognizeLinkedEntitiesResultCollection(recognizeLinkedEntitiesResultList, DEFAULT_MODEL_VERSION, textDocumentBatchStatistics);
+    }
 
-        LinkedEntity linkedEntity1 = new LinkedEntity(
-            "Seattle", new IterableStream<>(Collections.singletonList(linkedEntityMatch1)),
+    /**
+     * Helper method to get the expected linked Entities List 1
+     */
+    static List<LinkedEntity> getLinkedEntitiesList1() {
```
Comment (maririos) [inline sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java:236]: my concern with having all this tests to look at specific values is that if the service changes something about their model, then our test will fail.
Make sure whenever u call this that the model-version is set to avoid it

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java (hunk 5 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java:147]: To add getDetectedLanguageEnglish(), getDetectedLanguageSpanish(), and getUnknownDetectedLanguage() helper methods. These result repeatedly used over many place, such as atomic operation.

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java (hunk 6 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java:236]: Sounds good. Will take care of model-version in another PR. Since TA hasn't apply the model version in other APIs as well. 

issue created: #13943
